command/setup_consul.go

```diff
@@ -0,0 +1,449 @@
+`)
+
+	authMethodConf, err = s.renderAuthMethodConf(authMethodTasks)
+	if err != nil {
+		s.Ui.Error(err.Error())
+		return 1
+	}
+
+	jsConf, _ = json.MarshalIndent(authMethodConf, "", "\t")
+	s.Ui.Output(string(jsConf))
+
+	if !s.autoYes {
+		createAuthMethod = s.askQuestion(
+			fmt.Sprintf(
+				"Should we create the %s auth method in your Consul cluster? [Y/n]",
+				authMethodTasks,
+			))
+	} else {
+		createAuthMethod = true
+	}
+
+	if createAuthMethod {
+		err = s.createAuthMethod(authMethodTasks, authMethodConf)
+		if err != nil {
+			s.Ui.Error(err.Error())
+			return 1
+		}
+	}
+
+	s.Ui.Output(`
+In order to map claims between Nomad's JWTs and Consul ACL, we need to create
+the following binding rule:
+{
+	"Description": "binding rule for Nomad templates w/ (WI)",
+	"AuthMethod": "nomad-tasks",
+	"BindType": "role",
+	"BindName": "nomad-${value.nomad_namespace}-templates"
+}
+`)
+
+	if !s.autoYes {
+		createBindingRule = s.askQuestion(
+			"Should we create the above binding rule in your Consul cluster? [Y/n]",
+		)
+	} else {
+		createBindingRule = true
+	}
+
+	if createBindingRule {
+		err = s.createBindingRules(&api.ACLBindingRule{
+			Description: "binding rule for Nomad templates w/ (WI)",
+			AuthMethod:  authMethodTasks,
+			BindType:    "role",
+			BindName:    "nomad-${value.nomad_namespace}-templates",
+		})
+		if err != nil {
+			s.Ui.Error(err.Error())
+			return 1
+		}
+	}
+
+	s.Ui.Output(`
+Nomad tasks require a Consul ACL policy and Role. Below is the body of the policy
+we need to create:
+`)
+	s.Ui.Output(string(policyBody))
+
+	var createPolicy bool
+	if !s.autoYes {
+		createPolicy = s.askQuestion(
+			"Should we create the above policy in your Consul cluster? [Y/n]",
+		)
+	} else {
+		createPolicy = true
+	}
+
+	if createPolicy {
+		err = s.createPolicy()
+		if err != nil {
+			s.Ui.Error(err.Error())
+			return 1
+		}
+	}
+
+	s.Ui.Output(`
+Finally, we need to create a role role-nomad-tasks associated with the policy
+above.`)
+
+	var createRole bool
+	if !s.autoYes {
+		createRole = s.askQuestion(
+			"Should we create the above role in your Consul cluster? [Y/n]",
+		)
+	} else {
+		createRole = true
+	}
+
+	if createRole {
+		err = s.createRoleForTemplate()
+		if err != nil {
+			s.Ui.Error(err.Error())
+			return 1
+		}
+	}
+
+	s.Ui.Output(`
+Congratulations, your Consul cluster is now setup and ready to accept Nomad
+workloads with Workload Identity!
+
+You need to adjust your Nomad client configuration in the following way:
+
+consul {
+  enabled = true
+  address = "<Consul address>"
+
+  # Nomad agents still need a Consul token in order to register themselves
+  # for automated clustering. It is recommended to set the token using the
+  # CONSUL_HTTP_TOKEN environment variable instead of writing it in the
+  # configuration file.
+
+  # Consul mTLS configuration.
+  # ssl       = true
+  # ca_file   = "/var/ssl/bundle/ca.bundle"
+  # cert_file = "/etc/ssl/consul.crt"
+  # key_file  = "/etc/ssl/consul.key"
```

Comment:
I think it's ok to omit the mTLS config for this helper.

Reply:
But if they use mTLS then this config will be incomplete and not work. Since we have access to the Consul client config we know exactly what needs to be set in the file.

Using a go template here could be useful, so something like this (totally untested 😅):

```
consul {
  enabled = true
  address = "{{.Address}}"

  # Nomad agents still need a Consul token in order to register themselves
  # for automated clustering. It is recommended to set the token using the
  # CONSUL_HTTP_TOKEN environment variable instead of writing it in the
  # configuration file.

  {{if .Namespace)}}
  namemespace = "{{.Namespace}}"
  {{end}}

  {{if (eq .Scheme "https")}}
  ssl       = true
  {{end}}
  {{if .TLSConfig.CAFile}}
  ca_file   = "{{.TLSConfig.CAFile}}"
  {{end}}
  {{if .TLSConfig.CertFile}}
  cert_file = "{{.TLSConfig.CertFile}}"
  {{end}}
  {{if .TLSConfig.CertFile}}
  key_file  = "{{.TLSConfig.KeyFile}}"
  {{end}}
  {{if .TLSConfig.InsecureSkipVerify}}
  verify_ssl  = "{{.TLSConfig.KeyFile}}"
  {{end}}
}
```

And then you can just feed it the Consul client config to get the correct Nomad config file.

Bonus point if you run the output through [`hclwrite.Format()`](https://pkg.go.dev/github.com/hashicorp/hcl2@v0.0.0-20191002203319-fb75b3253c80/hclwrite#Format) 😄